Ref/tutorial data modification

joshuaunity opened the pull request:
## Description
This PR makes some modifications to the tutorial data as a follow up to he newly added flex-context aet field

## Look & Feel

![image](https://github.com/user-attachments/assets/80dd7034-acd6-4303-9a86-44ed5755e0df)

## How to test

Run the tutorial up to `Toy example II` all should go well without errors.

## Further Improvements

This isn't really about this PR, but this [page](http://localhost:5000/sensors/3) fails. When I looked through it,that page is trying to get data that I don't think exists

This function [get_breadcrumb_info](https://github.com/FlexMeasures/flexmeasures/blob/afeb40d309e77aa7607dd90a04915f441928061a/flexmeasures/ui/views/sensors.py#L90) is trying to get a data I don't think exist

## Related Items

Mention if this PR closes an Issue or Project.

This PR closes #1356 

---

- [x] I agree to contribute to the project under Apache 2 License. 
- [x] To the best of my knowledge, the proposed patch is not based on code under GPL or other license that is incompatible with FlexMeasures

joshuaunity commented:
@nhoening you can check both sugestions now

nhoening reviewed (changes requested):
Has `site-power-capacity` really transitioned to flex-context for everyone? In any case, I would move that change to a different PR. It doesn't seem to me that it is necessary for this PR.

Instead, I would remove the consumption price sensor from [the CLI command](https://github.com/FlexMeasures/flexmeasures/blob/main/documentation/tut/toy-example-from-scratch.rst?plain=1#L27), so there is no clash with the one you add on the db in this PR.

Also, please update the output of the show asset command [here](https://github.com/FlexMeasures/flexmeasures/blob/main/documentation/tut/toy-example-setup.rst?plain=1#L227)

joshuaunity commented:
>Instead, I would remove the consumption price sensor from [the CLI command](https://github.com/FlexMeasures/flexmeasures/blob/main/documentation/tut/toy-example-from-scratch.rst?plain=1#L27), so there is no clash with the one you add on the db in this PR.

So that will mean we remove the parameter and all its occurrences in that command.

nhoening commented:
> > Instead, I would remove the consumption price sensor from [the CLI command](https://github.com/FlexMeasures/flexmeasures/blob/main/documentation/tut/toy-example-from-scratch.rst?plain=1#L27), so there is no clash with the one you add on the db in this PR.
> 
> So that will mean we remove the parameter and all its occurrences in that command.

In that command, there is only 1 occurence?

Also update the docs [https://github.com/FlexMeasures/flexmeasures/blob/main/documentation/tut/toy-example-from-scratch.rst?plain=1#L22](here) if it works out.

joshuaunity commented:
> > > Instead, I would remove the consumption price sensor from [the CLI command](https://github.com/FlexMeasures/flexmeasures/blob/main/documentation/tut/toy-example-from-scratch.rst?plain=1#L27), so there is no clash with the one you add on the db in this PR.
> > 
> > 
> > So that will mean we remove the parameter and all its occurrences in that command.
> 
> In that command, there is only 1 occurence?
> 
> Also update the docs [https://github.com/FlexMeasures/flexmeasures/blob/main/documentation/tut/toy-example-from-scratch.rst?plain=1#L22](here) if it works out.

what I meant was that, inside the function itself, the variable is being used. do you mean we take those out as well

joshuaunity commented on flexmeasures/data/models/planning/storage.py:
```diff
@@ -975,7 +975,7 @@ def _get_device_power_capacity(
             power_capacity_in_mw = flex_model_d.get(
                 "power_capacity_in_mw",
-                sensor.get_attribute("capacity_in_mw"),
+                sensor.get_attribute("site-power-capacity"),
```
If I revert that, then that will mean I need to use "capacity_in_mw" on the toy flex_context cause as I am "site-power-capacity"

@nhoening

nhoening replied:
The idea in #1356 was not to use the flex context here, but a flex-model way of describing the *battery's* limits. *

So the `capacity_in_mw` instead of `site-power-capacity`.  As an attribute for now, because the flex-model does not exists as a db field just yet.

Maybe read the issue description again. Did this help?

* We'd still use `flex-context`, but for the consumption price.

joshuaunity replied:
Ive acutally figured this part out and its a mistake on my part, I put in the attributes wrongly so it wasn't seeing it, but it works now and this second erroris no more, I will revert the change here as well

nhoening commented:
> > > > Instead, I would remove the consumption price sensor from [the CLI command](https://github.com/FlexMeasures/flexmeasures/blob/main/documentation/tut/toy-example-from-scratch.rst?plain=1#L27), so there is no clash with the one you add on the db in this PR.
> > > 
> > > 
> > > So that will mean we remove the parameter and all its occurrences in that command.
> > 
> > 
> > In that command, there is only 1 occurence?
> > Also update the docs [https://github.com/FlexMeasures/flexmeasures/blob/main/documentation/tut/toy-example-from-scratch.rst?plain=1#L22](here) if it works out.
> 
> what I meant was that, inside the function itself, the variable is being used. do you mean we take those out as well

Which function? I need more help as I am diving in and out of this problem throughout the day and am missing/forgetting the context.

joshuaunity commented:
> Which function? I need more help as I am diving in and out of this problem throughout the day and am missing/forgetting the context.

Ok, let me giove more details

`flexmeasures add schedule for-storage --sensor 2 --consumption-price-sensor 1 --start ${TOMORROW}T07:00+01:00 --duration PT12H --soc-at-start 50% --roundtrip-efficiency 90%`

The above command gives errors when I run it, currently, there are two errors

### First Error:
```
marshmallow.exceptions.ValidationError: {'_schema': ['Must pass either consumption-price or consumption-price-sensor.']}
```

This error is happening because now in the toy asset, we added new value for the flex-context, which is ` flex_context={"consumption-price": {"sensor": day_ahead_sensor.id}},`

in this part of the [code](https://github.com/FlexMeasures/flexmeasures/blob/afeb40d309e77aa7607dd90a04915f441928061a/flexmeasures/data/models/planning/storage.py#L822) the scheduler merges its flex-context with the asset's flexcontext and the merged flex-context becomes

`{'consumption-price': {'sensor': 1}, 'consumption-price-sensor': 1, 'production-price-sensor': 1, 'inflexible-device-sensors': []}`

It then tries to validate this merged flexcontext with the FlexContextSchema, which throw an error because it does want both `consumption-price` and `consumption-price-sensor` to be together in one sensor, [see here](https://github.com/FlexMeasures/flexmeasures/blob/afeb40d309e77aa7607dd90a04915f441928061a/flexmeasures/data/schemas/scheduling/__init__.py#L165)

Potential solution:

We modify the key we added to the toy asset.

**FROM** 

flex_context={"consumption-price": {"sensor": day_ahead_sensor.id}},`

**TO**

flex_context={"consumption-price-sensor": {"sensor": day_ahead_sensor.id}},`

That way when the merge happens we don't get those too different keys

Another potential solution is that right at the [merge point](https://github.com/FlexMeasures/flexmeasures/blob/afeb40d309e77aa7607dd90a04915f441928061a/flexmeasures/data/models/planning/storage.py#L822) we check if this duplicate exist and remove one.

### The Second Error:

```
ValueError: Power capacity is not defined in the sensor attributes or the flex-model.
```

This is being thrown in this [function](https://github.com/FlexMeasures/flexmeasures/blob/afeb40d309e77aa7607dd90a04915f441928061a/flexmeasures/data/models/planning/storage.py#L959)

nhoening commented:
So, to repeat: I want to remove ` --consumption-price-sensor 1` from the CLI command. That should avoid the conflict.

joshuaunity commented:
>Also, please update the output of the show asset command [here](https://github.com/FlexMeasures/flexmeasures/blob/main/documentation/tut/toy-example-setup.rst?plain=1#L227)

I've updated this now

nhoening commented on flexmeasures/cli/data_add.py:
```diff
@@ -2125,11 +2134,13 @@ def create_asset_with_one_sensor(
+            flex_context={"consumption-price": {"sensor": day_ahead_sensor.id}},
+            unit="MW",
+            **{
                 "site-power-capacity": "500 kVA",
```
As I said in the issue, I wanted to stop using "site-power-capacity" and start using "capacity_in_mw". 

So it seems the tutorial was already showing what I want, while the code (here) was doing something else. I don't know why this mismatch came about, and it's good that this PR aligns both, but let's use the `capacity_in_mw` here (which later will become a way to show the new `flex_model` attribute when we have it (instead being just an attribute)

joshuaunity replied:
sorry i think i got it confused. I thought you didn't want it in the flex context but rather in the attributes. I'll change it.

nhoening replied:
Well, it has to be attributes also for now, as the flex-model is not there yet.

nhoening reviewed (changes requested):
Thanks.
Just some confusion about the asset attribute remaining.

joshuaunity commented on flexmeasures/data/models/planning/storage.py:
```diff
@@ -975,7 +975,7 @@ def _get_device_power_capacity(
             power_capacity_in_mw = flex_model_d.get(
                 "power_capacity_in_mw",
-                sensor.get_attribute("capacity_in_mw"),
+                sensor.get_attribute("site-power-capacity"),
```
this will also reverted

joshuaunity commented on documentation/tut/toy-example-setup.rst:
```diff
@@ -224,9 +224,9 @@ If you want, you can inspect what you created:
-    battery  (52.374, 4.88969)  capacity_in_mw: 0.5
+    Type     Location           Attributes                    Flex-Context
+    -------  -----------------  ----------------------------  ---------------------------- 
+    battery  (52.374, 4.88969)  site-power-capacity: 0.5      consumption-price: {'sensor': 1}
```
Yes you, are right I had a feeling there was one more  place I was missing out, thanks for spotting this it would've been confusing for anyone working on this

joshuaunity commented on flexmeasures/cli/data_add.py:
```diff
@@ -2125,11 +2134,13 @@ def create_asset_with_one_sensor(
-                "site-power-capacity": "500 kVA",
+            flex_context={"consumption-price": {"sensor": day_ahead_sensor.id}},
+            unit="MW",
+            **{
```
Yeah we can, I can revert that.

joshuaunity commented:
@nhoening we wil need to update the test cli to remove the `conusmption-price-sensor` arg as its causing and error and the test is filing

nhoening commented:
Yes, that is basically the same change, just not in the tutorial but in the script used in building the docker image in GitHub Actions. It basically tests running the first tutorial part.

You can find that by using some text search like `grep`. 

I don't know by heart where that is.

Let me know if you can't find it.

joshuaunity commented:
Hi @nhoening. Although I removed the `consumption` arg, the test still fails cause it say the arg is needed. but when I run on my local it behaves as expected.

On the github action env it acts as if the arg is `required`

nhoening commented:
That seems strange. Maybe I have an idea later, but right now not.
Can you repeat the job?

nhoening reviewed (changes requested):
About the failing test: I don't see where you would make the requirement of that sensor optional. Are you sure it works locally ?

I thought we agreed that when calling `tb_utils.replace_deprecated_argument()` for the "consumption-price-sensor" (that is where this error appears), we can now add the parameter `required_argument=False`.

That should really solve this error.

Or am I overlooking where that error is now avoided?

joshuaunity commented:
>I thought we agreed that when calling tb_utils.replace_deprecated_argument() for the "consumption-price-sensor" (that is where this error appears), we can now add the parameter required_argument=False.

This is et to false already by default

joshuaunity commented:
> About the failing test: I don't see where you would make the requirement of that sensor optional. Are you sure it works locally ?
> 
> I thought we agreed that when calling `tb_utils.replace_deprecated_argument()` for the "consumption-price-sensor" (that is where this error appears), we can now add the parameter `required_argument=False`.
> 
> That should really solve this error.
> 
> Or am I overlooking where that error is now avoided?

This is my result from running it locally

```
flexmeasures add schedule for-storage --sensor 2 --start 2025-04-12T07:00+01:00 --duration PT12H --soc-at-start 50% --roundtrip-efficiency 90%
[FLEXMEASURES][2025-04-15 21:13:13,370] INFO: Loading config from /home/joshua-edward/Documents/repos/seita_projects/flexmeasures/instance/flexmeasures.cfg ...
Warning: Without complete mail settings, FlexMeasures will not be able to send mails to users, e.g. for password resets!
Warning: Without complete redis connection settings, FlexMeasures will not be able to run forecasting and scheduling job queues.
You might consider setting MAIL_USERNAME, MAIL_PASSWORD, FLEXMEASURES_REDIS_PASSWORD.
[FLEXMEASURES][2025-04-15 21:13:13,370] INFO: [FLEXMEASURES] No SENTRY_DSN setting found, so initialising Sentry cannot happen ...
[FLEXMEASURES][2025-04-15 21:13:14,058] INFO: Loaded plugins: {}
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/venv/lib/python3.12/site-packages/timely_beliefs/beliefs/utils.py:1183: FutureWarning: BeliefsDataFrame.fillna with 'method' is deprecated and will raise in a future version. Use obj.ffill() or obj.bfill() instead.
  df = df.fillna(
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/venv/lib/python3.12/site-packages/timely_beliefs/beliefs/utils.py:1183: FutureWarning: BeliefsDataFrame.fillna with 'method' is deprecated and will raise in a future version. Use obj.ffill() or obj.bfill() instead.
  df = df.fillna(
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1373: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  storage_device_constraints["min"] = storage_device_constraints["min"].fillna(
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1386: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  storage_device_constraints["max"] = storage_device_constraints["max"].fillna(
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1603: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  lhs = constraints_df.fillna(0).eval(lhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1604: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  rhs = constraints_df.fillna(0).eval(rhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1603: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  lhs = constraints_df.fillna(0).eval(lhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1604: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  rhs = constraints_df.fillna(0).eval(rhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1603: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  lhs = constraints_df.fillna(0).eval(lhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1604: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  rhs = constraints_df.fillna(0).eval(rhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1603: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  lhs = constraints_df.fillna(0).eval(lhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1604: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  rhs = constraints_df.fillna(0).eval(rhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1603: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  lhs = constraints_df.fillna(0).eval(lhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1604: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  rhs = constraints_df.fillna(0).eval(rhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1603: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  lhs = constraints_df.fillna(0).eval(lhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1604: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  rhs = constraints_df.fillna(0).eval(rhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1603: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  lhs = constraints_df.fillna(0).eval(lhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1604: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  rhs = constraints_df.fillna(0).eval(rhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1603: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  lhs = constraints_df.fillna(0).eval(lhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1604: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  rhs = constraints_df.fillna(0).eval(rhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1603: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  lhs = constraints_df.fillna(0).eval(lhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1604: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  rhs = constraints_df.fillna(0).eval(rhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1603: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  lhs = constraints_df.fillna(0).eval(lhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1604: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  rhs = constraints_df.fillna(0).eval(rhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1603: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  lhs = constraints_df.fillna(0).eval(lhs_expression).round(round_to_decimals)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/flexmeasures/data/models/planning/storage.py:1604: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  rhs = constraints_df.fillna(0).eval(rhs_expression).round(round_to_decimals)
[FLEXMEASURES][2025-04-15 21:13:14,250] INFO: [FLEXMEASURES][2025-04-15 21:13:14,243] INFO: [FLEXMEASURES][2025-04-15 21:13:14,243] INFO: Running HiGHS 1.8.0 (git hash: eda5cbe): Copyright (c) 2024 HiGHS under MIT licence terms
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/venv/lib/python3.12/site-packages/timely_beliefs/beliefs/classes.py:633: SAWarning: Object of type <TimedBelief> not in session, add operation along 'Sensor.beliefs' will not proceed (This warning originated from the Session 'autoflush' process, which was invoked automatically in response to a user-initiated operation.)
  df = pd.DataFrame(session.execute(q))
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/venv/lib/python3.12/site-packages/timely_beliefs/beliefs/classes.py:633: SAWarning: Object of type <TimedBelief> not in session, add operation along 'DataSource.beliefs' will not proceed (This warning originated from the Session 'autoflush' process, which was invoked automatically in response to a user-initiated operation.)
  df = pd.DataFrame(session.execute(q))
[FLEXMEASURES][2025-04-15 21:13:14,287] INFO: SAVING TO DB...
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/venv/lib/python3.12/site-packages/timely_beliefs/beliefs/classes.py:332: SAWarning: Object of type <TimedBelief> not in session, add operation along 'Sensor.beliefs' will not proceed (This warning originated from the Session 'autoflush' process, which was invoked automatically in response to a user-initiated operation.)
  session.execute(smt)
/home/joshua-edward/Documents/repos/seita_projects/flexmeasures/venv/lib/python3.12/site-packages/timely_beliefs/beliefs/classes.py:332: SAWarning: Object of type <TimedBelief> not in session, add operation along 'DataSource.beliefs' will not proceed (This warning originated from the Session 'autoflush' process, which was invoked automatically in response to a user-initiated operation.)
  session.execute(smt)
New schedule is stored.
```

nhoening commented:
> > I thought we agreed that when calling tb_utils.replace_deprecated_argument() for the "consumption-price-sensor" (that is where this error appears), we can now add the parameter required_argument=False.
> 
> This is set to false already by default

Not according to [what I see here](https://github.com/SeitaBV/timely-beliefs/blob/main/timely_beliefs/utils.py#L184).

Also, this is a parameter I would pass by name (`required_argument=False`) to be more explicit.

nhoening commented on flexmeasures/cli/data_add.py:
```diff
@@ -1389,6 +1389,7 @@ def add_schedule_for_storage(  # noqa C901
         optimization_context_sensor,
         "consumption-price-sensor",
         consumption_price_sensor,
+        False,
```
```suggestion
        required_argument=False,
```

nhoening commented:
> > > > I thought we agreed that when calling tb_utils.replace_deprecated_argument() for the "consumption-price-sensor" (that is where this error appears), we can now add the parameter required_argument=False.
> > > 
> > > 
> > > This is set to false already by default
> > 
> > 
> > Not according to [what I see here](https://github.com/SeitaBV/timely-beliefs/blob/main/timely_beliefs/utils.py#L184).
> > Also, this is a parameter I would pass by name (`required_argument=False`) to be more explicit.
> 
> The test is now passing, the function already had the value set to false by default I had no idea it wanted me to still explicitly state it, I just tried it after reading your previous coming just to be sure, and turns out that was the problem.
> 
> But strangely enough it works fine in my local even without this modification

Where was this default value? I pointed to the code that uses `True` as default...
